compiler/damlc/daml-stdlib-src/DA/Internal/LF.daml

```diff
@@ -216,3 +217,7 @@ newtype AnyTemplate = AnyTemplate { getAnyTemplate : Any }
 
 -- | Existential choice type that can wrap an arbitrary chice.
 newtype AnyChoice = AnyChoice { getAnyChoice : Any }
+
+-- | Unique textual representation of a template Id.
+newtype TemplateTypeRep = TemplateTypeRep { getTemplateTypeRep : Text }
+  deriving (Eq, Ord)
```

Comment:
We probably want a `MapKey` instance for this which we can easily get via GND but we can leave that for a separate PR.